modules/core/include/opencv2/core/directx.hpp

```diff
@@ -129,6 +133,16 @@ CV_EXPORTS void convertToD3D11Texture2D(InputArray src, ID3D11Texture2D* pD3D11T
 //! @param dst             - destination OutputArray
 CV_EXPORTS void convertFromD3D11Texture2D(ID3D11Texture2D* pD3D11Texture2D, OutputArray dst);
 
+#ifdef HAVE_OPENCL
+//! @brief Converts ID3D11Texture2D to cl_mem (OpenCL image2d_t).
+//
+//! @note Note: Function creates an OpenCL 2D image object from a subresource of a Direct3D 11 2D texture
+//
+//! @param pD3D11Texture2D - source D3D11 texture
+//! @param pointer to OpenCL image2d_t object
+CV_EXPORTS void convertFromD3D11Texture2DtoCLImage(ID3D11Texture2D* pD3D11Texture2D, cl_mem* clImage);
```

Comment:
i believe  we shoudl avoid using HAVE_OPENCL define when exporting interface functions from library

one way is remove define and `cl_mem` and pass `void*` instead. but i think some thin wrapper would be better: may be it has presented in openCV already

Reply:
Removed